Eagle-Rock-Analytics/historical-obs-platform
# QA/QC: accumulated precipitation variables logic check

This PR adds a new QAQC function and modifies a few things for clarity

**QAQC function added**: qaqc_precip_logic_accum_amounts performs 2 checks:
- checks the accumulated precipitation vars (pr_5min, pr_1h, pr_24h) to ensure that accumulation in a shorter window does not exceed accumulation in a longer window, i.e., pr_5min is not larger than pr_1h
- checks that pr_localmid is not larger than pr_24h (this catches spikes/dips in pr_24h)

The era_qaqc_flag_meanings.csv was updated with a new column that identifies which QA/QC function placed the flags; greater transparency and check over process. 

---
**To test**
Run ALLNETWORKS_qaqc.py with a network of choice and watch for the "14", "15", and "16" flags that will be printed in the `Precip eraqc flags (any other value than nan is an active flag!)` print statements. These will be a rare occurrence, but if you want to see a station where these checks do occur, run this script by doing the following:

For the first check (multi-vars):
- comment out Ln 77
- set Ln 78 to be ` for station in ['RAWS_SMOC1']:`
- set network = "RAWS" in Ln 275

For the second check (localmid):
- comment out Ln 77
- set Ln 78 to be `for station in ['CWOP_UKBKB']:`
- set network = "CWOP" in Ln 275

Additionally/alternatively: look through the qaqc_testing.ipynb notebook, which reads in a dummy precipitation file, with some notes for what is being tested and what should be flagged

---
**QUESTIONS**
1. qaqc_precip_logic_accum_amounts has three flags, one for above value, one for below, and one specifically for when pr_24h is lower than pr_localmid. Do we standardize to one since by this process to a single above/below flag? 
2. Do we keep the qaqc testing notebook and dummy precipitation file, and if so, does it live in this folder or elsewhere?
